### PR TITLE
Update EIP-3670: Drop "terminating" instruction restriction

### PR DESCRIPTION
**Pull request by chfast**
Drop the requirement that the last instruction in code is "terminating". The change requires adding explicit rule about immediate data.

### CODE REVIEW

File: EIPS/eip-3670.md [hunk 1]
```diff
@@ -44,23 +44,19 @@ The EOF1 format provides following forward compatibility properties:
 
 This feature is introduced on the very same block EIP-3540 is enabled, therefore every EOF1-compatible bytecode MUST be validated according to these rules.
 
-Previously deprecated instructions `CALLCODE` (0xf2) and `SELFDESTRUCT` (0xff) are invalid and their opcodes do not have any instruction assigned.
+1. Previously deprecated instructions `CALLCODE` (0xf2) and `SELFDESTRUCT` (0xff) are invalid and their opcodes do not have any instruction assigned.
 
-At contract creation time both *initcode* and *code* are iterated instruction-by-instruction (the same process is used to perform "JUMPDEST-analysis").
-Bytecode is deemed invalid if any of these conditions is true:
+2. At contract creation time *instructions validation* is performed on both *initcode* and *code*. The code is invalid if any of the checks below are failed. For each instruction:
```
Comment (gumb0) [inline EIPS/eip-3670.md:49]:
```suggestion
2. At contract creation time *instructions validation* is performed on both *initcode* and *code*. The code is invalid if any of the checks below fails. For each instruction:
```

File: EIPS/eip-3670.md [hunk 2]
```diff
@@ -44,23 +44,19 @@ The EOF1 format provides following forward compatibility properties:
...
+   1. Check if the opcode is defined.
```
Comment (gumb0) [inline EIPS/eip-3670.md:50]: Maybe keep the note about `INVALID`?

File: EIPS/eip-3670.md [hunk 3]
```diff
@@ -87,19 +81,18 @@ Each case should be tested for creation transaction, `CREATE` and `CREATE2`.
...
 - EOF code containing undefined instruction
 - EOF code ending with incomplete `PUSH` instruction
     - This can include `PUSH` instruction unreachable by execution, e.g. after `STOP`
-- EOF code ending with any defined instruction other than terminating ones
-    - This can include codes where non-terminating instruction is not reachable, e.g. `0030` (`STOP ADDRESS`).
 
 ## Reference Implementation
 
 ```python
+from eip3540 import ValidationException
```
Comment (gumb0) [inline EIPS/eip-3670.md:94]: I omitted this import from other eips, I think it only takes space

**Review by daliruination1 — APPROVED** [review]
[]()

File: EIPS/eip-3540.md [hunk 4]
```diff
@@ -125,7 +125,7 @@ For clarity, the *container* refers to the complete account code, while *code* r
 
 1. `JUMPDEST`-analysis is only run on the *code*.
 2. Execution starts at the first byte of the *code*, and `PC` is set to 0.
-3. If `PC` goes outside the code section bounds, execution aborts with failure.
+3. Execution stops if `PC` goes outside the code section bounds.
```
Comment (lightclient) [inline EIPS/eip-3540.md:128]: @axic: should update motivation to not mention benefit of knowing last instruction is terminating.

File: EIPS/eip-3670.md (hunk 3 above)
Comment (chfast) [reply EIPS/eip-3670.md]: Done.

**Pull request merged**